clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -639,19 +645,22 @@ private void maybeUpdate(long now, Node node) {
                 ClientRequest clientRequest = request(now, nodeConnectionId, metadataRequest);
                 log.debug("Sending metadata request {} to node {}", metadataRequest, node.id());
                 doSend(clientRequest, now);
-            } else if (connectionStates.canConnect(nodeConnectionId, now)) {
+            } else if ((lastConnectAttemptNodeId != null && isConnectionClosed(lastConnectAttemptNodeId) ||
```

Comment:
Wouldn't it make more sense to check `lastConnectAttemptNodeId` in `maybeUpdate(long now)` before finding a new node with `leastLoadedNode`?